File: SobrecargaMetodos/Soma.cs
using System;

public class MinhaClasse
{
    public int Soma(int a, int b)
    {
        return a + b;
    }

    public string Soma(string a, string b)
    {
        return a + b;
    }

    public static void Main(string[] args)
    {
        MinhaClasse minhaInstancia = new MinhaClasse();

        // Soma de inteiros
        int somaInt = minhaInstancia.Soma(5, 3);
        Console.WriteLine("Soma de inteiros: " + somaInt);

        // Concatenação de strings
        string concatStr = minhaInstancia.Soma("Hello", " World");
        Console.WriteLine("Concatenação de strings: " + concatStr);
    }
}

File: SobrecargaMetodos/Soma.cpp
#include <iostream>
#include <string>

class MinhaClasse {
public:
    int soma(int a, int b) {
        return a + b;
    }

    std::string soma(std::string a, std::string b) {
        return a + b;
    }
};

int main() {
    MinhaClasse minhaInstancia;

    // Soma de inteiros
    int somaInt = minhaInstancia.soma(5, 3);
    std::cout << "Soma de inteiros: " << somaInt << std::endl;

    // Concatenação de strings
    std::string concatStr = minhaInstancia.soma("Hello", " World");
    std::cout << "Concatenação de strings: " << concatStr << std::endl;

    return 0;
}

